dialog/base.go

```diff
@@ -108,8 +108,13 @@ func (d *dialog) hideWithResponse(resp bool) {
 func (d *dialog) create(buttons fyne.CanvasObject) {
 	label := widget.NewLabelWithStyle(d.title, fyne.TextAlignLeading, fyne.TextStyle{Bold: true})
 
+	image := &canvas.Image{Resource: d.icon}
```

Comment:
If the dialog is immutable once created (ie if it doesn't have an image it will never have one), why not avoid adding the image entirely, or if it's easier code-wise, place a `layout.NewSpacer()` in place of the image in the container, as it's a less expensive object. That's what I often use as an "empty" canvas object.

Reply:
You're right it was a code complexity issue that led me to leave it there.
My concern is that by placing a spacer we have possibly different objects in that index - is a spacer much less expensive than a hidden image with nil content?

Reply:
I checked `canvas.Image` implementation and the only reason I even bothered with the suggestion is that Resize on an image results in a Refresh: https://github.com/fyne-io/fyne/blob/master/canvas/image.go#L174

Reply:
I am fine with merging this as-is though (but I didn't actually test it myself) and we could file an optimization ticket to the backlog if we want.

Reply:
> I checked `canvas.Image` implementation and the only reason I even bothered with the suggestion is that Resize on an image results in a Refresh

You talked me into it thank :) spacer is actually OK - all tests and usages passing